content/en/docs/architecture/images.md

```diff
@@ -0,0 +1,124 @@
+
+* a PR to the component repository modifies the `FROM` directive in `Dockerfile`. CI jobs on this pull request will reflect the use of the new Go version and repository administrators are expected to monitor the change, merging it if appropriate or landing changes as necessary to merge the change themselves. After code freeze in the release, any un-updated `FROM` entries become blocker bugs for the release. If an OCP component cannot migrate to the new toolchain version, the owner has to get the approval from the architecture team and then communicate with the ART team. In this case, the ART team will modify the builder image of the component in `ocp-build-data` and the created PR to the component repository can be closed. For example, `cluster-etcd-operator` has to use `golang-1.14`. `images/cluster-etcd-operator.yml` will has the following stanza:
+
+  ```yaml
+  from:
+    builder:
+    - stream: golang-1.14
+  ```
+
+* a PR to the `openshift/release` repository modifies the CI configuration to allow builds with the new Go version to be as efficient as possible. This PR is merged automatically.
+
+{{< alert title="Info" color="info" >}}
+To avoid triggering a thundering herd of CI jobs when the Go toolchain changes centrally for OCP, pull requests to component repositories that change `FROM` entries are made slowly. It may take up to a day to complete this process for all repositories.
+{{< /alert >}}
+
+The productized images are always built with the builder images defined in ocp-build-data, no matter whether component authors merge pull requests that change `FROM` entries.
+
+As two pull requests are created to reflect changes in the Go toolchain, it is possible for them to merge in either order. The CI system will be self-consistent and correct in either case. When only one of the changes (to `FROM` entries in the component repository and to `ci-operator` configuration in `openshift/release`) has merged, container image builds in CI will use the correct (newer) Go toolchain, but will simply be less efficient than they might be. This intermediate state will have an inconsistency between the base_images entry in ci-operator configuration and the `FROM` entries in the repository but this may be ignored.
+
+## Best Practice
+
+* Merge the PR to the component repo as soon as possible to use the new version of GoLang toolchain.
+* Do not override manually the replacement rules of the builder and base images in CI configuration and let automation do its job.
+* Only manage ocp-build-data for base and builder images.
+* Configure `build_root` using via `.ci-operator.yaml` in the component repo (see `build_root.from_repository` in [the documentation](/docs/architecture/ci-operator/#build-root-image)). Although it is not enforced, it is suggested to use the same version of `GoLang` in `build_root` as the builder image.
```

Comment:
This should IMO be it's own separate section. For users who *need* some non-standard version of Go or who *must* change the version of Go in their repos while also changing other content in the repo, this is a valid approach. This approach answers the question of "how do I control the version of Go used in my repository" or similar